### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by SaltScratch**
Allows users to build and run xbscli using docker instead of Go - 

git clone https://github.com/mrusme/xbscli.git
cd xbscli
docker build -t xbscli-docker .
docker run --rm xbscli-docker -i "xbs id" -p "xbs key" -s "xbs url" -f pretty > bookmarks.txt

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM golang:1.16-alpine
```
Comment (mrusme) [inline Dockerfile]: You might want to change this, see https://github.com/mrusme/xbscli/blob/9aa886e7b19e9214ba2e9046fee8d363e2689c77/go.mod#L3C1-L3C8

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+WORKDIR /app
+
+COPY go.mod ./
+COPY go.sum ./
+RUN go mod download
+
+COPY lz77/*.go ./lz77/
+COPY *.go ./
```
Comment (mrusme) [inline Dockerfile]: Why not simply `ADD . .` instead of the four individual `COPY` commands?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+RUN go build -o /xbscli-docker
```
Comment (mrusme) [inline Dockerfile]: Why would it require a dedicated output name over using the default (`xbscli`)?

Comment (SaltScratch) [reply Dockerfile]: fixed

File: Dockerfile (hunk 2 above)
Comment (SaltScratch) [reply Dockerfile]: fixed

File: Dockerfile (hunk 1 above)
Comment (SaltScratch) [reply Dockerfile]: updated to 1.19

**Comment by SaltScratch** [comment]
As requested I have also updated the readme. I included both the docker and html format stuff

**Comment by mrusme** [comment]
Thank you very much, great work!

**Pull request merged**